core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -33,21 +33,40 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
       .filter(_.completed != requestedIncomplete)
     val allAppsSize = allApps.size
 
+    val eventLogsUnderProcessCount = parent.getEventLogsUnderProcess()
+    val lastUpdatedTime = parent.getLastUpdatedTime()
     val providerConfig = parent.getProviderConfig()
     val content =
+      <script src={UIUtils.prependBaseUri("/static/historypage-common.js")}></script>
       <div>
           <div class="span12">
             <ul class="unstyled">
               {providerConfig.map { case (k, v) => <li><strong>{k}:</strong> {v}</li> }}
             </ul>
             {
+            if (eventLogsUnderProcessCount > 0) {
+              <p>There are {eventLogsUnderProcessCount} event log(s) currently being
+                processed which may result in additional applications getting listed on this page.
+                Refresh the page to view updates. </p>
+            }
+            }
+
+            {
+            if (lastUpdatedTime > 0) {
+              <p>Last updated: <span id="last-updated">{lastUpdatedTime}</span></p>
```

Comment:
Is this just going to display a Long to the user? That's not going to be very informative. What about a human time where <= 0 is mapped to "never"?